From 99f64a538f6b4f58022af8b1162a51bfa03601e3 Mon Sep 17 00:00:00 2001
From: Jian-Ming Wang <jian.ming.wang@mitaccomputing.com>
Date: Mon, 12 Jan 2026 09:17:16 +0000
Subject: [PATCH 1/1] Added back formatter for StringView to fix build isssue.

Upstream-Status: Inappropriate [No upstream maintenance]
---
 include/boost_formatters.hpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/boost_formatters.hpp b/include/boost_formatters.hpp
index e675c3b1..bf8397fa 100644
--- a/include/boost_formatters.hpp
+++ b/include/boost_formatters.hpp
@@ -52,4 +52,16 @@ struct std::formatter<UrlBase>
     }
 };
 
+template <std::derived_from<boost::core::string_view> StringView>
+struct std::formatter<StringView>
+{
+    constexpr auto parse(std::format_parse_context& ctx)
+    {
+        return ctx.begin();
+    }
+    auto format(const StringView& msg, auto& ctx) const
+    {
+        return std::format_to(ctx.out(), "{}", std::string_view(msg));
+    }
+};
 // NOLINTEND(readability-convert-member-functions-to-static, cert-dcl58-cpp)
-- 
2.45.2

