From a8e76f7ceb18d0908366fac1a72a99c6f1afd704 Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Wed, 23 Apr 2025 06:20:51 +0000
Subject: [PATCH 1/1] Fixed build issue from casting.

Upstream-Status: Inappropriate [No upstream maintenance]
---
 drivers/soc/aspeed/ast2600-espi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/aspeed/ast2600-espi.c b/drivers/soc/aspeed/ast2600-espi.c
index 283c1eabbc46..f01b6b0ff649 100644
--- a/drivers/soc/aspeed/ast2600-espi.c
+++ b/drivers/soc/aspeed/ast2600-espi.c
@@ -1880,7 +1880,7 @@ static int ast2600_espi_flash_probe(struct ast2600_espi *espi)
 
 	of_property_read_u32(dev->of_node, "flash-safs-mode", &flash->safs.mode);
 	if (flash->safs.mode == SAFS_MODE_MIX) {
-		rc = of_property_read_u32(dev->of_node, "flash-safs-tgt-addr", &flash->safs.taddr);
+		rc = of_property_read_u32(dev->of_node, "flash-safs-tgt-addr", (u32*)&flash->safs.taddr);
 		if (rc || !IS_ALIGNED(flash->safs.taddr, FLASH_SAFS_ALIGN)) {
 			dev_err(dev, "cannot get 16MB-aligned SAFS target address\n");
 			return -ENODEV;
-- 
2.25.1

